skimmer/skimmer_utils.py

```diff
@@ -493,62 +493,76 @@ def apply_variation(events, variation):
             )
 
         return events
-    
+
+
+def __add_weight_variations(events, variation_up, variation_down, nominal, name):
+    """Add the up/down weight branches to the events."""
+
+    # Normalize the weights such that the sum of nominal weights
+    # can be used to normalize the events for the variations
+    variation_up = variation_up / ak.mean(variation_up, axis=0)
+    variation_down = variation_down / ak.mean(variation_down, axis=0)
```

Comment:
This is what was done in the other code to normalize to unit cross-section, no?
Does this only accidentally normalizes the PDF weights and it should be different in this case?

Reply:
No you are right, I just got confused :)

Reply:
Thinking twice, this is a "terrible" idea: if you divide by the mean, you divide by sumw_variation / N, meaning dividing by sumw_variation and multiplying by N, where N is the number of MC events. And N = sumw_nominal because the nominal weights are 1...
So it works only by "coincidence" because the gen weights are 1... Better do the proper rescaling by sumw_nominal / sumw_variation!